### DISCUSSION TITLE
Problems trying to create dynamic input.http source

### DISCUSSION THREAD

**Discussion by Russsgithub** [savonet/liquidsoap#1887] [Q&A] [answered]
```
settings.init.force_start.set(false)
set("server.telnet", true)
set("log.level",5)

sources = ref([])

safe = playlist("/home/russ/files")
safe_source = buffer(mksafe(safe))

def play(pl)
  #s = mksafe(playlist("playlists/#{pl}"))
  mount = "testv2"
  print("Recieved request for http stream from - #{pl}")
  s = mksafe(input.http("#{pl}"))
  s = output.icecast(%mp3,fallible = true, host="host", port=8000, password="pass", mount="#{mount}", s)
  sources := list.add((pl, s), !sources)
  "Playing #{pl}."
end

def stop(pl)
  s = list.assoc(pl, !sources)
  sources := list.assoc.remove(pl, !sources)
  s.shutdown()
  "Stopped #{pl}."
end

server.register("play", play)
server.register("stop", stop)

output.dummy(mksafe(safe_source))
```

I'm using this code in an attempt to create an input.http source via telnet, but get an ffmpeg error
```
2021/09/08 10:20:07 [input.http_0:4] Connection failed: FFmpeg.Avutils.Error(No such file or directory)
```

If I use a static url in the play method then I get a connection and the print "#{pl}" line returns the correct url in quotes when i issue the play "url" through telnet.

Is it possible to do this at present ?

Thanks

**Comment by Russsgithub** [comment]
I can get this to work. I think the incoming stream may have hung at the time of testing.

**Comment by toots** [comment, marked as answer]
Hi,

That looks like a pretty cool script. It works locally actually, when used with this url: `https://supersoul.live:8020/SRRS320kbps`.

Could you post more logs or perhaps a URL that does not work for you?

**Comment by Russsgithub** [comment]
Hi @toots 

Yes, it works for me too. ( comment above yours ). I've been running liq inside screen and using copy mode to scroll logs, I think i left copy mode on and the script hung, so i was getting silence using this script and thought i had issues.

... though .. is there a way to interact with the play and stop commands via a web interface rather than telnet ?

Thanks

**Reply by toots** [reply]
Yes! See here: https://www.liquidsoap.info/doc-dev/harbor_http.html

**Reply by Russsgithub** [reply]
Great. Thanks @toots
